File: 1331-RankTransformofanArray/1331-RankTransformofanArray.py
# Last updated: 8/8/2026, 2:35:26 PM
1class Solution:
2    def arrayRankTransform(self, arr: List[int]) -> List[int]:
3        rank_map = {}
4        rank = 1
5
6        for num in sorted(arr):
7            if num not in rank_map:
8                rank_map[num] = rank
9                rank += 1
10        return [rank_map[num] for num in arr]
11        
12        '''n = int(input.strip())
13        arr = list(map(int, input.split()))
14        result = rankofint(arr)
15        return (result)'''

File: 1331-RankTransformOfAnArray/1331-RankTransformOfAnArray.cpp
// Last updated: 9/6/2026, 12:31:31 PM
class Solution {
public:
    vector<int> arrayRankTransform(vector<int>& arr) {
        vector<int> temp = arr;
        int res = 0;
        sort(temp.begin(), temp.end());
        int n = arr.size();

        temp.erase(unique(temp.begin(), temp.end()), temp.end());



        for(int i = 0; i<n; i++){
            arr[i] = lower_bound(temp.begin(), temp.end(), arr[i]) - temp.begin() + 1;
        }
        return arr;
    }
};